proxy/certs/certs.go

```diff
@@ -215,9 +217,29 @@ func (s *RemoteCertSource) Remote(instance string) (cert *x509.Certificate, addr
 		logging.Errorf("%v", err)
 		logging.Errorf("WARNING: specifying the correct region in an instance string will become required in a future version!")
 	}
-	if len(data.IpAddresses) == 0 || data.IpAddresses[0].IpAddress == "" {
+
+	if len(data.IpAddresses) == 0 {
 		return nil, "", "", fmt.Errorf("no IP address found for %v", instance)
 	}
+
+	//Add the option to choose which IP address to use based on user input ip_address_type
+	//Assume IP address type order is not fixed, we need to find the corresponding IP address by iterating IPAddresses
+	userIpAddressType := flag.Lookup("ip_address_type").Value.String()
+	userChosenIpAddress := ""
+	errorMessageIpAddress := ""
+	for _, eachIpAddressOption := range data.IpAddresses {
+		errorMessageIpAddress += "(TYPE: " + eachIpAddressOption.Type + " IP: " + eachIpAddressOption.IpAddress + ") "
+		if(strings.ToLower(eachIpAddressOption.Type) == strings.ToLower(userIpAddressType)) {
```

Comment:
Consider adding "PUBLIC" as an alias for "PRIMARY": see listIpTypes in https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/pull/78/commits/fda79905410f489bcafed5eaed6a8731965ef355